### PR TITLE
feat(map_based_prediction): enable to control lateral path convergence speed by setting control  time horizon

### PR DESCRIPTION
**Pull request by YoshiRi**
## Description

In the current path generation algorithm, lateral displacement is controlled so that the it will arrive center line after the `time_horizon_` = `10sec`. This causes slow lateral motion convergence and lead to inappropriate path outside the lanelet.

I just add one parameter to control this convergence speed. Like figure below.
It simply means to calculate lateral path which converge to centerline by faster time_horizon.

image

Must merge after https://github.com/autowarefoundation/autoware_launch/pull/637.

## Tests performed

Tested with Psim.

See jira ticket [TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-2371).

## Notes for reviewers

**Must change `autoware_launch` repo before merging this to main branch**.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/map_based_prediction/config/map_based_prediction.param.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@
   ros__parameters:
     enable_delay_compensation: true
     prediction_time_horizon: 10.0 #[s]
+    lateral_control_time_horizon: 5.0 #[s]
```
Comment (kyoichi-sugahara) [inline perception/map_based_prediction/config/map_based_prediction.param.yaml:5]: @YoshiRi 
could you add explanation in the README.md?

File: perception/map_based_prediction/src/path_generator.cpp [hunk 2]
```diff
@@ -213,16 +214,20 @@ FrenetPath PathGenerator::generateFrenetPath(
...
-    const double d_next = current_point.d + current_point.d_vel * t + 0 * 2 * std::pow(t, 2) +
-                          lat_coeff(0) * std::pow(t, 3) + lat_coeff(1) * std::pow(t, 4) +
-                          lat_coeff(2) * std::pow(t, 5);
+    const double d_next_ = current_point.d + current_point.d_vel * t + 0 * 2 * std::pow(t, 2) +
+                           lat_coeff(0) * std::pow(t, 3) + lat_coeff(1) * std::pow(t, 4) +
+                           lat_coeff(2) * std::pow(t, 5);
+    // t > lateral_duration: 0.0, else d_next_
+    const double d_next = t > lateral_duration ? 0.0 : d_next_;
     const double s_next = current_point.s + current_point.s_vel * t + 2 * 0 * std::pow(t, 2) +
                           lon_coeff(0) * std::pow(t, 3) + lon_coeff(1) * std::pow(t, 4);
```
Comment (kyoichi-sugahara) [inline perception/map_based_prediction/src/path_generator.cpp:232]: @YoshiRi
Is it needed to keep the coefficient 0 term?

File: perception/map_based_prediction/config/map_based_prediction.param.yaml (hunk 1 above)
Comment (kyoichi-sugahara) [reply perception/map_based_prediction/config/map_based_prediction.param.yaml:5]: [NITS]
`time_to_center_line` is easier for me to understand the feature.

**Comment by kyoichi-sugahara** [comment]
@YoshiRi 
LGTM!
Are you going to run scenario test?

**Comment by kyoichi-sugahara** [comment]
@YoshiRi 
Great feature, thank you very much! 
One thing I was thinking is that while it's inevitable that the trajectory will expand with the jerk-minimized trajectory, 
it might be difficult to tune the parameter for the convergence time of the lateral deviation. 
Would it be possible to determine the convergence time of the lateral deviation based on the maximum jerk instead?

**Comment by YoshiRi** [comment]
@kyoichi-sugahara 
Test passed in [Evaluator](https://evaluation.tier4.jp/evaluation/reports/32875021-16d9-52de-bac8-fd19681b40db?project_id=prd_jt).

> it might be difficult to tune the parameter for the convergence time of the lateral deviation.
Would it be possible to determine the convergence time of the lateral deviation based on the maximum jerk instead?

It's more simple. "Jerk minimized path" finally means quintic polynominal equations calculated with boundary conditions and we can just tune convergence time by changing ending boundarys.

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**